src/main/java/com/invoiced/entity/Note.java

```diff
@@ -54,7 +62,19 @@ protected long getEntityId() {
 	@Override
 	@JsonIgnore
 	protected String getEntityName() {
-		return "notes";
+		if (this.currentOperation == "listAll" && this.subAssociation != null) {
+			return listAllEntityName();
+			}
+		else {
+			return "notes";
+		}
+	}
+
+	@JsonIgnore
+	private String listAllEntityName() {
+		if (this.subAssociation == "invoice") return "invoices/" + String.valueOf(this.invoiceId) + "/notes";
```

Comment:
Is there a reason for making the Note class aware of customers and invoices instead of adding listNotes() methods to the Customer and Invoice entities?

Reply:
My reasoning in doing it this way is to make it analogous to how contacts are handled, i.e. being able to get a list of them using `customer.newNote().listAll()`. Without this addition, the above line would unexpectedly retrieve a list of all notes on any customer.

Reply:
You can also add a getter and setter for the list endpoint and simplify this logic so that Notes are not aware of Customer or Invoice endpoints